homeassistant/components/caldav/config_flow.py

```diff
@@ -0,0 +1,173 @@
+"""Config flow to configure caldav."""
+import logging
+
+import voluptuous as vol
+
+from homeassistant import config_entries
+from homeassistant.const import CONF_PASSWORD, CONF_URL, CONF_USERNAME, CONF_VERIFY_SSL
+from homeassistant.core import callback
+from homeassistant.data_entry_flow import FlowResult
+from homeassistant.helpers import selector
+import homeassistant.helpers.config_validation as cv
+
+from . import async_caldav_connect
+from .const import (
+    CALDAV_EXCEPTIONS,
+    CONF_ADD_CUSTOM_CALENDAR,
+    CONF_CALENDARS,
+    CONF_CUSTOM_CALENDARS,
+    CONF_DAYS,
+    DEFAULT_VERIFY_SSL,
+    DOMAIN,
+)
+
+DATA_SCHEMA = vol.Schema(
+    {
+        # pylint: disable=no-value-for-parameter
+        vol.Required(CONF_USERNAME): selector.TextSelector(),
+        vol.Required(CONF_PASSWORD): selector.TextSelector(
+            selector.TextSelectorConfig(type=selector.TextSelectorType.PASSWORD)
+        ),
+        vol.Required(CONF_URL): selector.TextSelector(
+            selector.TextSelectorConfig(type=selector.TextSelectorType.URL)
+        ),
+        vol.Optional(
+            CONF_VERIFY_SSL, default=DEFAULT_VERIFY_SSL
+        ): selector.BooleanSelector(),
+        vol.Optional(CONF_DAYS, default=1): cv.positive_int,
+        vol.Optional(CONF_CALENDARS, default=[]): selector.SelectSelector(
+            selector.SelectSelectorConfig(options=[], custom_value=True, multiple=True),
+        ),
+        vol.Required(CONF_ADD_CUSTOM_CALENDAR, default=False): bool,
+    }
+)
+
+CUSTOM_CALENDARS_SCHEMA = vol.Schema(
+    {
+        vol.Optional(
+            CONF_CUSTOM_CALENDARS,
+            description={
+                "suggested_value": [
+                    {
+                        "name": "HomeOffice",
+                        "calendar": "Agenda",
+                        "search": "HomeOffice",
+                    },
+                    "# Please clean this example",
```

Comment:
Ideally we don't have English text inline in the field since it's not translatable